### PR TITLE
feat: add 2 hour granularity for bigquery and druid

### PR DESCRIPTION
**Pull request by eschutho**
### SUMMARY
Adds 2 hour time granularity for bigquery and druid
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
For BigQuery
Data: 
Screenshot_3_17_21__6_57_PM

With time granularity:
_DEV__Explore_-_cases_ts_03_17_2021_18_36_16

UI with BigQuery: 
_DEV__Explore_-_cases_ts_03_17_2021_19_09_53

### TEST PLAN
I could use help testing on Druid from someone who has access to a db

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
@betodealmeida

**Comment by betodealmeida** [comment]
Looks great! I'm spinning up a Druid service to test the PR.

**Comment by etr2460** [comment]
What happens on non-BigQuery/Druid datasets? Does the option not show up? Or does it show up but not work?

**Comment by betodealmeida** [comment]
> What happens on non-BigQuery/Druid datasets? Does the option not show up? Or does it show up but not work?

@etr2460 the option doesn't show up, each DB engine spec defines the time grains it supports.

**Comment by betodealmeida** [comment]
Tested with Druid:

Screenshot_2021-03-17 Explore - wikipedia

**Review by betodealmeida — APPROVED** [review]

**Comment by etr2460** [comment]
@betodealmeida even though it lives in controls.jsx too? weird... So the list in controls.jsx needs to be a superset of all time granularities that are supported across all db engine specs? And if someone wants to add a custom db engine spec with custom time grains, it won't format properly on the client?

obviously not blocking this PR, but that's _weird_

File: superset-frontend/src/explore/controls.jsx
Comment (etr2460) [inline superset-frontend/src/explore/controls.jsx:283]: do you think these should be plural or singular? I see both being used (`5 minutes` but also `6 hour`). Maybe we can standardize while you're in here?

File: superset/db_engine_specs/base.py
Comment (etr2460) [inline superset/db_engine_specs/base.py:89]: here also uses singular for `5 minute` `10 minute` `15 minute` etc.

**Review by etr2460 — APPROVED** [review]
lgtm with one grammar nit, up to you if you want to address

File: superset-frontend/src/explore/controls.jsx
Comment (eschutho) [reply superset-frontend/src/explore/controls.jsx:283]: I saw the inconsistencies as well and thought it safest to leave them alone for now, but agree that it would be nice to fix them. Do you mind if I put up a separate or for that?

**Comment by eschutho** [comment]
@villebro left some comments in another thread about the possibility of the granularity list looking too long. Just want to make sure that he has a chance to comment before this is merged. Thanks all.

File: superset-frontend/src/explore/controls.jsx
Comment (villebro) [reply superset-frontend/src/explore/controls.jsx:283]: I'm not familiar with how these strings are used, but should we be translating these with `tn()`?

**Review by villebro — APPROVED** [review]
LGTM. In the long term, we should introduce a method in `BaseEngineSpec` that takes as an input an ISO-8601 duration and returns a SQL expression. This would make it possible to implement arbitrary granularities where supported (later we could make the time grain component in the UI support this).

File: superset-frontend/src/explore/controls.jsx
Comment (etr2460) [reply superset-frontend/src/explore/controls.jsx:283]: a separate PR is good, thanks!

Comment (eschutho) [reply superset-frontend/src/explore/controls.jsx:283]: That's a good point @villebro. I'll add some translations in the next PR, too. thanks!

**Comment by eschutho** [comment]
I made one small change rn that was directly related to my code, and seemed the simplest fix with regard to capitalization and singular/plural form.
Screenshot_3_18_21__10_18_AM

**Comment by eschutho** [comment]
If this looks good to you all for now, then 🏷 ready-to-merge.
I'll follow up with more changes to the strings (consistency and translations) in a separate PR, and I may have a need to add a 6 hour granularity, but then will start to explore the customization piece.

**Comment by villebro** [comment]
> If this looks good to you all for now, then 🏷 ready-to-merge.
> I'll follow up with more changes to the strings (consistency and translations) in a separate PR, and I may have a need to add a 6 hour granularity, but then will start to explore the customization piece.

Sounds good 👍  I'm happy to help out with the follow-up work @eschutho , I think this should be possible to break up into pretty bitesize chunks and I'd love to see the possibility of adding support for arbitrary time intervals.

**Comment by eschutho** [comment]
Thanks all for the review. I'm going to close this PR and use the configs for this instead. I've tested and it seems to be working. I'll still follow up on the other fixes and we can talk @villebro about the more sophisticated mapping, too.

**Pull request closed without merging**
